pkg/agent/protocol/http/proxy_test.go

```diff
@@ -371,35 +313,54 @@ func Test_ProxyHandler(t *testing.T) {
 		t.Run(tc.title, func(t *testing.T) {
 			t.Parallel()
 
-			client := &fakeHTTPClient{
-				body:    tc.body,
-				status:  tc.expectedStatus,
-				headers: tc.headers,
-			}
+			upstreamServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+				for k, values := range tc.headers {
+					for _, v := range values {
+						rw.Header().Add(k, v)
+					}
+				}
+				rw.WriteHeader(tc.statusCode)
 
-			upstreamURL, err := url.Parse(tc.config.UpstreamAddress)
+				_, err := rw.Write(tc.body)
+				if err != nil {
+					t.Errorf("writing upstream body: %v", err)
+				}
+			}))
+
+			upstreamURL, err := url.Parse(upstreamServer.URL)
 			if err != nil {
-				t.Errorf("error parsing upstream address %v", err)
-				return
+				t.Fatalf("error parsing httptest url")
 			}
+
 			handler := &httpHandler{
 				upstreamURL: *upstreamURL,
-				client:      client,
+				client:      http.DefaultClient,
```

Comment:
`client` was introduced only for testing. If we are now always using the DefaultClient, maybe we could simply remove this field.